modules/scheduler/gke-cluster/main.tf

```diff
@@ -62,6 +62,19 @@ locals {
     "SYSTEM_COMPONENTS",
     "WORKLOADS"
   ]
+
+  required_permissions = [
+    "resourcemanager.projects.setIamPolicy",
+    "container.clusters.create",
+    "container.clusters.delete",
+    "compute.instances.create",
+    "compute.instances.delete",
+    "storage.buckets.create",
+    "iam.serviceAccounts.create",
+    "iam.serviceAccounts.setIamPolicy",
+    "iam.serviceAccounts.actAs",
+    "serviceusage.services.use"
+  ]
```

Comment:
Are all the roles related to storage and serviceAccounts going to be checked for in the cluster module?

Is there a plan to split the tests to the relevant modules? For example, the storage module will test whether the user has roles to create the storage. In this case, the error will still not show up until the deployment starts working with the storage module.

Reply:
+1 We should make this piece into a generic data structure so that any list of modules can be checked. Something along the lines of a map with `key=moduleName` and `value=list[]` where the list contains all the permissions. Then re-use this generic data structure through the blueprint. Feel free to put together a short design document to flesh out the idea and alternatives to it.